src/state/swap/hooks.ts

```diff
@@ -235,27 +239,24 @@ export function useDerivedSwapInfo(platformOverride?: RoutablePlatform): UseDeri
     }
 
     // Use a static version
-    const staticProvider = provider ? new StaticJsonRpcProvider(provider?.connection.url) : undefined
-
-    const getTrades = isExactIn
-      ? getExactInFromEcoRouter(
-          {
-            currencyAmountIn: parsedAmount,
-            currencyOut: outputCurrency,
-            ...commonParams,
-          },
-          ecoRouterSourceOptionsParams,
-          staticProvider
-        )
-      : getExactOutFromEcoRouter(
-          {
-            currencyAmountOut: parsedAmount,
-            currencyIn: inputCurrency,
-            ...commonParams,
-          },
-          ecoRouterSourceOptionsParams,
-          staticProvider
-        )
+    const staticProvider = provider
+      ? provider
+          .getNetwork()
+          .then(network => (provider ? new StaticJsonRpcProvider(provider?.connection.url, network) : undefined))
+      : Promise.resolve(undefined)
```

Comment:
This can be all replaced with:

```suggestion
        const staticProvider = provider ? new StaticJsonRpcProvider(provider.connection, provider.network) : undefined
```

The promise part is necessary.